Release notes and attribute update for MTV 2.11.3 (#896)
kubev2v/forklift-documentation: documentation/modules/rn-2-11-3-resolved-issues.adoc

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

:_content-type: REFERENCE
[id="resolved-issues-2-11-3_{context}"]
= Resolved issues 2.11.3

[role="_abstract"]
Review the resolved issues in this release of {project-short}.

Storage copy offload: Cryptic error message for PowerMax migrations replaced::

> Before this update, during storage copy offload migrations with Dell PowerMax storage, if an ESXi host's Fibre Channel or iSCSI initiators were not registered as a host object in PowerMax, the code proceeded with an empty `hostID`. This resulted in a cryptic `CreateMaskingView` error message: `Parameter value is not valid, '' is not facet-valid with respect to pattern '[a-zA-Z0-9_-]+' for type 'HostGroupIdTypeVer2'.`

Reviewer: ```suggestion
Before this update, if an ESXi host's Fibre Channel or iSCSI initiators were not registered as a host object in Dell PowerMax during a storage copy offload migration, the migration proceeded with an empty `hostID` value. As a consequence, the system displayed an unclear `CreateMaskingView` error message. With this update, the system validates the initiator registration. As a result, if the initiators are not registered, the system displays the following message to help you resolve the problem:
```

With this release, if the problem occurs, the following message appears, telling you how to fix the problem:  

`Can't find a host on symmetrix %s with initiators matching %v. Ensure the ESXi host has a corresponding host object in PowerMax with the correct FC/iSCSI initiators registered.`
+
link:https://issues.redhat.com/browse/MTV-4873[MTV-4873]

Storage copy offload: Pure Storage FlashArray HTTP client timeout is now configurable::

> Before this update, the timeout for a Pure Storage FlashArray HTTP client was hardcoded to 30 seconds. As a result, during migrations of VMs with many disks, simultaneous `CopyVolume` requests to the Pure Storage FlashArray timed out, leaving PVCs stuck in `Pending` status. With this update, you can configure the timeout by setting the new `STORAGE_HTTP_TIMEOUT_SECONDS` key and value in the Pure Storage FlashArray `Secret`` Note that the key’s default value is 30 seconds. 

Reviewer: ```suggestion
Before this update, large (multi-terabyte) warm migrations took a long time during the `fstrim` stage of `virt-v2v`. With this update, {project-short} uses the `--no-fstrim` option for `virt-v2v`. As a result, large warm migrations run correctly and finish faster. Note that the converted guest might use more space after the conversion.
```

These changes strictly adheres to the Cause-Consequence-Fix-Result (CCFR) structural template mandated by the Red Hat Supplementary Style Guide for "Fixed issues" (bug fixes).

1. Misuse of Transition Phrases (CCFR Logic)The Red Hat style guide requires a very specific flow for bug fixes:Cause: "Before this update, <cause>." Consequence: "As a consequence, <consequence>." Fix: "With this update, <fix>." Result: "As a result, <result>." In your first snippet, you used "As a result" to describe the problem ("As a result, during migrations... requests... timed out"). The style guide reserves "As a result" exclusively for describing the positive outcome after the patch is applied. You should use "As a consequence" to describe the negative user experience caused by the bug.

2. Missing the "Result" StatementThe first snippet explains the cause, the consequence, and the fix, but it completely drops the final required "Result" statement. The guide requires you to explicitly state what happens now that the patch is applied (e.g., "As a result, migrations complete successfully without timing out.").

3. Formatting and TyposThe first snippet contains a formatting typo at the very end: ...in the Pure Storage FlashArray Secret`` Note that... (it uses a double backtick and misses a period).

Author: @anarnold97 I've modified the text of the release note based on the comments above. The suggested text  was a repeat of the previous item, so please review the new text.

+
link:https://issues.redhat.com/browse/MTV-5004[MTV-5004]

Large warm migrations now proceed quicker in the `fstrim` stage of `virt-v2v`::

> Before this update, some very large (multi-terabyte) warm migrations took a very long time at the `fstrim` stage of `virt-v2v`. With this update, {project-short} uses the `--no-fstrim` option for `virt-v2v`, allowing large warm migrations to run as expected. This might result in the converted guest using more space after conversion.

Reviewer: ```suggestion
Large warm migrations no longer stall during the `fstrim` stage of `virt-v2v`::
Before this update, large (multi-terabyte) warm migrations took a long time during the `fstrim` stage of `virt-v2v`. With this update, {project-short} uses the `--no-fstrim` option for `virt-v2v`. As a result, large warm migrations run correctly and finish faster. Note that the converted guest might use more space after the conversion.
```

* Strict Adherence to CCFR Structure: The changes properly apply the mandatory Cause-Consequence-Fix-Result (CCFR) tense logic required for "Fixed issues".

    * Fix: "With this update, {project-short} uses the --no-fstrim option...".
    
    * Result: It separates the outcome into its own distinct sentence using the required transition phrase: "As a result, large warm migrations run correctly and finish faster.". The first version blurred the fix and the result together.
    
* Removal of Prohibited Phrases ("as expected"): The first version states the migrations "run as expected." Expectations are relative, and the Red Hat Supplementary Style Guide explicitly forbids the phrase "as expected". The second version correctly replaces this with the objective adverb "correctly".

* Elimination of Fluff (Minimalism): The first version relies on unnecessary qualifiers ("some very large", "very long time"). The second version deletes these empty modifiers ("large", "long time"), adhering to guidelines that require succinct, direct writing and the elimination of fluff.

* Active Voice and Directness: The first version uses a clunky, passive construction for the final warning ("This might result in the converted guest using more space..."). The second version ("Note that the converted guest might use more space...") uses the active voice and directly addresses the subject, which improves readability and translation.

// Module included in the following assemblies:
//
// * documentation/doc-Release_notes/master.adoc

:_content-type: REFERENCE
[id="resolved-issues-2-11-3_{context}"]
= Resolved issues 2.11.3

[role="_abstract"]
Review the resolved issues in this release of {project-short}.

Storage copy offload: Unclear error message for PowerMax migrations replaced::

Before this update, if an ESXi host's Fibre Channel or iSCSI initiators were not registered as a host object in Dell PowerMax during a storage copy offload migration, the migration proceeded with an empty `hostID` value. As a  consequence, the system displayed an unclear `CreateMaskingView` error message.
+
With this update, the system validates the initiator registration. As a result, if the initiators are not registered, the system displays the following message to help you resolve the problem:
+

> `Can't find a host on symmetrix %s with initiators matching %v. Ensure the ESXi host has a corresponding host object in PowerMax with the correct FC/iSCSI initiators registered.`

Reviewer: ```suggestion
`Can't find a host on Symmetrix %s with initiators matching %v. Ensure the ESXi host has a corresponding host object in PowerMax with the correct FC/iSCSI initiators registered.`
```
Symmetrix is a Dell proprietary brand

+
link:https://issues.redhat.com/browse/MTV-4873[MTV-4873]

> Storage copy offload: Pure Storage FlashArray HTTP client timeout is now configurable::

Reviewer: ```suggestion
Storage copy offload: Pure Storage FlashArray HTTP client timeout is configurable::
```

Before this update, the timeout for a Pure Storage FlashArray HTTP client was hard-coded to 30 seconds. As a consequence, during migrations of VMs with many disks, simultaneous `CopyVolume` requests to the Pure Storage FlashArray timed out, leaving PVCs stuck in `Pending` status.
+
With this update, you can configure the timeout by setting the new `STORAGE_HTTP_TIMEOUT_SECONDS` key and value in the Pure Storage FlashArray `Secret`. As a result, migrations complete successfully without timing out. Note that the key’s default value is 30 seconds.
+
link:https://issues.redhat.com/browse/MTV-5004[MTV-5004]

The `fstrim` stage of `virt-v2v` is now skipped for warm migrations::

Before this update, large (multi-terabyte) warm migrations took a long time during the `fstrim` stage of `virt-v2v`.
+

> With this update, {project-short} uses the `--no-fstrim` option for `virt-v2v`. As a result, large warm migrations run correctly. Note that the converted guest might use more space after the conversion.

Reviewer: ```suggestion
With this update, {project-short} uses the `--no-fstrim` option for `virt-v2v`. As a result, large warm migrations run correctly. The converted guest might use more space after the conversion.
```
